cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js

```diff
@@ -459,9 +464,9 @@ describe('discover_table', () => {
           .then((noEntries) => {
             cy.setTopNavDate(
               'Sep 22, 2015 @ 14:00:00.000',
-              'Sep 22, 2015 @ 18:00:00.000'
+              'Sep 22, 2015 @ 14:30:00.000'
             );
-            cy.waitForLoader();
```

Comment:
why did you change this? wait for loader is more reliable since it waits for a specific element and not just the an arbitrary time which can cause the test to become flakey

Reply:
While running the tests, I found that waiting for page load didn't make sense in few occasions. While applying the filter to a table the entire page doesn't reload. In those scenarios instead of waiting for the number of entires to change , cypress picks up the older result and the evaluation condition fails. 
To avoid such scenarios I have put a wait time.

Reply:
@ashwin-pc Updated the PR to remove waits, instead added verification for hits which has resolved the previous issues.